File: src/main/java/com/theagilemonkeys/workshop/controllers/SearchController.java
package com.theagilemonkeys.workshop.controllers;

import com.theagilemonkeys.ellmental.core.schema.Embedding;
import com.theagilemonkeys.workshop.services.EmbeddingsSpaceService;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.ResponseBody;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
public class SearchController {
    private final EmbeddingsSpaceService embeddingsSpaceService;

    public SearchController(EmbeddingsSpaceService embeddingsSpaceService) {
        this.embeddingsSpaceService = embeddingsSpaceService;
    }

    @GetMapping("/search")
    @CrossOrigin
    @ResponseBody
    public List<Embedding> search(@RequestParam String query, @RequestParam(defaultValue = "10") int itemsLimit) {
        // TODO: implement the search functionality using the EmbeddingsSpaceService
        throw new UnsupportedOperationException("TODO");
    }
}

File: workshop-kotlin/src/main/kotlin/com/theagilemonkeys/workshop/controllers/SearchController.kt
package com.theagilemonkeys.workshop.controllers

import com.theagilemonkeys.ellmental.semanticsearch.SearchOutput
import com.theagilemonkeys.workshop.services.SemanticSearchService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController


@RestController
class SearchController(@Autowired private val semanticSearchService: SemanticSearchService) {
    @GetMapping("/search")
    suspend fun search(@RequestParam query: String, @RequestParam itemsLimit: Int = 10): SearchOutput {
        // TODO: implement the search functionality using the SemanticSearchService
        throw UnsupportedOperationException("Not implemented")
    }
}